join_group_request_test.go

```diff
@@ -0,0 +1,62 @@
+package sarama
+
+import "testing"
+
+var (
+	joinGroupRequestNoProtocols = []byte{
+		0, 9, 'T', 'e', 's', 't', 'G', 'r', 'o', 'u', 'p', // Group ID
+		0, 0, 0, 100, // Session timeout
+		0, 0, // Member ID
+		0, 8, 'c', 'o', 'n', 's', 'u', 'm', 'e', 'r', // Protocol Type
+		0, 0, 0, 0, // 0 protocol groups
+	}
+
+	joinGroupRequestOneProtocol = []byte{
+		0, 9, 'T', 'e', 's', 't', 'G', 'r', 'o', 'u', 'p', // Group ID
+		0, 0, 0, 100, // Session timeout
+		0, 11, 'O', 'n', 'e', 'P', 'r', 'o', 't', 'o', 'c', 'o', 'l', // Member ID
+		0, 8, 'c', 'o', 'n', 's', 'u', 'm', 'e', 'r', // Protocol Type
+		0, 0, 0, 1, // 1 group protocol
+		0, 3, 'o', 'n', 'e', // Protocol name
+		0, 0, 0, 3, 0x01, 0x02, 0x03, // protocol metadata
+	}
+
+	joinGroupRequestTwoProtocols = []byte{
+		0, 9, 'T', 'e', 's', 't', 'G', 'r', 'o', 'u', 'p', // Group ID
+		0, 0, 0, 100, // Session timeout
+		0, 12, 'T', 'w', 'o', 'P', 'r', 'o', 't', 'o', 'c', 'o', 'l', 's', // Member ID
+		0, 8, 'c', 'o', 'n', 's', 'u', 'm', 'e', 'r', // Protocol Type
+		0, 0, 0, 2, // 2 group protocol
+		0, 3, 't', 'w', 'o', // Protocol name
+		0, 0, 0, 3, 0x04, 0x05, 0x06, // protocol metadata
+		0, 3, 'o', 'n', 'e', // Protocol name
+		0, 0, 0, 3, 0x01, 0x02, 0x03, // protocol metadata
```

Comment:
This gets assembled from a map, so the order is not deterministic. @eapache how have you gotten around this elsewhere?